### DISCUSSION TITLE
Notification with progress bar

### DISCUSSION THREAD

**Discussion by mlarcher** [mantinedev/mantine#1626] [Feature requests]
I already said it elsewhere, but this library is truly amazing ! Thanks again !!

Simple question: is there a way to someow have a progressbar in a Notification to represent the time left before it closes ? And a way to pause the autoclose when the mouse hovers the notification ? That would be really great :)

**Comment by rtivital** [comment]
Hi Matthieu, thanks for kind words! We currently do not provide this feature (and probably never will) – it does not make much sense from the UX standpoint: user's attention was already attracted by notification appear animation, there is no point in distracting user with notification timeout. 

If you need this feature in your application, I can only recommend building your own notification with Affix, Dialog or Noficiation components.

**Comment by dan-lee** [comment]
I am using a combination of the notification system and a [`RingProgress`](https://mantine.dev/core/ring-progress).
It's not super optimized for animations or anything, but check out this CodeSandbox: https://codesandbox.io/s/fragrant-hill-bwre1z

<img width="446" alt="image" src="https://user-images.githubusercontent.com/571589/194745701-6228f144-fb08-40f7-93c7-dd40112cc85a.png">


With that, it should not be too hard to add a pause functionality.
